examples/lead-generation/src/graph/nodes/url-validator-node.ts

```diff
@@ -0,0 +1,135 @@
+import { getAirtopClient } from "@/airtop-client";
+import type { Url, UrlOutput, UrlState } from "@/graph/state";
+import { URL_VALIDATOR_OUTPUT_SCHEMA } from "@/graph/state";
+import type { BatchOperationError, BatchOperationInput, BatchOperationResponse } from "@airtop/sdk";
+import { getLogger } from "@local/utils";
+import { zodToJsonSchema } from "zod-to-json-schema";
+
+// Name of the edge
+export const VALID_URL_COUNTER_EDGE = "valid-url-counter-edge";
+
+/**
+ * Conditional edge that determines if the graph should continue to the next node based on valid URLs or go to the error handler
+ * @param state - The state of the URL node
+ * @returns The next edge to take
+ */
+export const validUrlCounterEdge = (state: UrlState): string => {
+  const validUrlCount = state.urls.filter((url) => !!url?.isValid).length;
+
+  return validUrlCount > 0 ? "fetch-therapists" : "error-handler";
+};
+
+/**
+ * Validates a string to determine if it is a valid URL
+ * @param url - The URL to validate
+ * @returns Whether the URL is valid
+ */
+const validateString = (url: string): boolean => {
```

Comment:
We could use a library like https://www.npmjs.com/package/validator or similar to verify if this is a valid url